modules/diagnostics.py

```diff
@@ -5,28 +5,46 @@
 from commands2 import CommandScheduler
 from wpilib import RobotController
 
+from commands.diagnostics.claw import DiagnoseClaw
+from commands.diagnostics.intake import DiagnoseIntake
+from commands.diagnostics.diagnoseall import DiagnoseAll
+from commands.diagnostics.utils.setrunningtest import SetRunningTest
 from modules.hardware import HardwareModule
 from ultime.module import Module, ModuleList
 
 
 class DiagnosticsModule(Module):
     def __init__(self, hardware: HardwareModule, module_list: ModuleList):
         super().__init__()
-        self.components_tests: List[commands2.Command] = []
+        self.components_tests: List[commands2.Command] = [
+            DiagnoseIntake(hardware.intake),
+            DiagnoseClaw(hardware.claw),
+        ]
 
         self._hardware = proxy(hardware)
         self._module_list = proxy(module_list)
         self._battery_voltage: List[float] = []
         self._is_test = False
 
+        self._run_all_command = DiagnoseAll(
+            self._hardware,
+            [
+                SetRunningTest(next(iter(component.getRequirements())), True)
+                .andThen(component)
+                .andThen(SetRunningTest(next(iter(component.getRequirements())), False))
+                for component in self.components_tests
+            ],
+        )
```

Comment:
Je ne sais pas ce que tu trouves qui est plus "clean looking" entre ta version actuelle et ça, je te laisse décider ce que tu préfères.

```suggestion
subsystems = [next(iter(component.getRequirements())) for component in components]

self._run_all_command = DiagnoseAll(
    self._hardware,
    [
        sequence(
            SetRunningTest(subsystem, True),
            component,
            SetRunningTest(subsystem, False)
        )
        for subsytem, component in zip(subsystem, self.components_tests)
    ],
)
```